nj2k/src/org/jetbrains/kotlin/nj2k/conversions/DefaultArgumentsConversion.kt

```diff
@@ -91,18 +92,31 @@ class DefaultArgumentsConversion(context: NewJ2kConverterContext) : RecursiveApp
                 .zip(calledMethod.parameters)
                 .drop(method.parameters.size)
 
+            fun JKSymbol.isNeedThisReceiver(): Boolean {
+                val parameters = defaults.map { it.second }
+                val declarations = element.declarations
+                return parameters.any { it.name.value == this.name } && declarations.any { it == this.target }
+            }
 
             for ((defaultValue, parameter) in defaults) {
                 fun remapParameterSymbol(on: JKTreeElement): JKTreeElement {
+                    if (on is JKQualifiedExpression && on.receiver is JKThisExpression) {
+                        return on
+                    }
                     if (on is JKFieldAccessExpression) {
                         val target = on.identifier.target
                         if (target is JKParameter) {
                             val newSymbol =
                                 symbolProvider.provideUniverseSymbol(calledMethod.parameters[method.parameters.indexOf(target)])
                             return JKFieldAccessExpression(newSymbol)
                         }
+                        if (on.identifier.isNeedThisReceiver()) {
+                            return JKQualifiedExpression(JKThisExpression(JKLabelEmpty(), JKNoType), JKFieldAccessExpression(on.identifier))
+                        }
+                    }
+                    if (on is JKCallExpression && on.identifier.isNeedThisReceiver()) {
```

Comment:
Why do we need this check? As far as I know, there should be no conflict with parameter/function names

In the following example `x` is resolved fine to the function `x`:
```
fun x() = 10
fun f(x: Int, y: Int = x()) = 42
```

Reply:
This is necessary for `isD: Boolean = this.isD` in:
https://github.com/t-kameyama/kotlin/blob/dfdf07a5cf21fcc557a5e19cce0a4f933a34c7bf/nj2k/testData/newJ2k/overloads/ConflictParameterName.kt#L23

Reply:
You are right :) In that way let's conver the similar case when `getSmth` becomes `smth` like:

```
public class Test {
    int getX() { return 1; }

    void foo() { foo(getX()); }
    void foo(int x) {}
}
```
Becomes
```
class Test {
    internal val x: Int internal get() = 1
    @JvmOverloads
    internal fun foo(x: Int = x/*error*/) {}
}
```

Reply:
Fixed